posts/2021-03-24-whats-new-in-MP-Rest-Client2.0.adoc

```diff
@@ -0,0 +1,212 @@
+class. For example, suppose you need to access an endpoint via a proxy server at myproxy.xyz.com on port 1080. You could
+build your rest client instance with code like:
+
+[source,java]
+----
+MyClient client = RestClientBuilder.newBuilder()
+                                   .proxyAddress(myproxy.xyz.com, 1080)
+                                   //...
+                                   .build(MyClient.class);
+----
+
+Alternatively, you can specify the proxy address via MP Config with a property like:
+
+`com.mypkg.MyClient/mp-rest/proxyAddress=myproxy.xyz.com:1080`
+
+Note that this approach to setting the proxy server host and port should be preferred to using vendor-specific
+properties such as `com.ibm.ws.jaxrs.client.proxy.host` and `com.ibm.ws.jaxrs.client.proxy.port`, though these
+properties will still work. For proxy authentication, you can still use the `com.ibm.ws.jaxrs.client.proxy.username` and
+`com.ibm.ws.jaxrs.client.proxy.password` properties. For example:
+[source,java]
+----
+MyClient client = RestClientBuilder.newBuilder()
+                                   .proxyAddress(myproxy.xyz.com, 1080)
+                                   .property("com.ibm.ws.jaxrs.client.proxy.username", "andymc12")
+                                   .property("com.ibm.ws.jaxrs.client.proxy.password", "12345") //same as my luggage! :)
+                                   //...
+                                   .build(MyClient.class);
+----
+
+[#auto-redirects]
+== Automatically following redirects
+
+If a RESTful resource has been relocated, often the HTTP response code will be in the 300 range and will indicate the
+new location. Rather than handling the 3XX response and manually issuing a new request, MP Rest Client 2.0 allows rest
+client instances to automatically follow redirects. This can be configured programmatically while building the client
+instance or via MP Config. Here is an example of configuring auto-redirect via the `RestClientBuilder` API:
+
+[source,java]
+----
+MyClient client = RestClientBuilder.newBuilder()
+                                   .followRedirects(true)
+                                   //...
+                                   .build(MyClient.class);
+----
+
+And here is how you would configure it via MP Config:
+
+`com.mypkg.MyClient/mp-rest/followRedirects=true`
+
+[#sse-support]
+== Support for Server Sent Events (SSEs)
+
+link:https://openliberty.io/guides/reactive-messaging-sse.html[Server Sent Events] are part of the HTML 5 spec and
+enables a server to push data to a client asynchronously via events - all over HTTP. The JAX-RS 2.1 spec enabled SSE
+support for both the client and server. Now you can consume SSE events from the type-safe MP Rest Client.
+
+The MP Rest Client specification uses the link:http://www.reactive-streams.org/[Reactive Streams] APIs to consume
+events. A client interface capable of consuming SSEs would look something like this:
```

Comment:
A client interface capable of consuming SSEs would look something like this:

--->

A client interface capable of consuming SSEs looks something like this: